File: src/main/java/krasa/editorGroups/actions/ViewPanelToggleAction.java
package krasa.editorGroups.actions;

import com.intellij.openapi.actionSystem.ActionUpdateThread;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.ToggleAction;
import com.intellij.openapi.project.DumbAware;
import krasa.editorGroups.EditorGroupsSettingsState;
import krasa.editorGroups.PanelRefresher;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class ViewPanelToggleAction extends ToggleAction implements DumbAware {
  @Override
  public @NotNull ActionUpdateThread getActionUpdateThread() {
    return ActionUpdateThread.BGT;
  }

  public ViewPanelToggleAction() {
    super("Editor Groups Panel");
  }

  @Override
  public boolean isSelected(@NotNull AnActionEvent event) {
    return EditorGroupsSettingsState.state().isShowPanel();
  }

  @Override
  public void setSelected(@NotNull AnActionEvent event, boolean state) {
    EditorGroupsSettingsState editorGroupsSettingsState = EditorGroupsSettingsState.state();
    editorGroupsSettingsState.setShowPanel(!editorGroupsSettingsState.isShowPanel());
    PanelRefresher.getInstance(Objects.requireNonNull(event.getProject())).refresh();
  }
}

File: src/main/java/krasa/editorGroups/actions/ViewPanelToggleAction.kt
package krasa.editorGroups.actions

import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.ToggleAction
import com.intellij.openapi.project.DumbAware
import krasa.editorGroups.EditorGroupsSettingsState.Companion.state
import krasa.editorGroups.PanelRefresher

class ViewPanelToggleAction : ToggleAction("Editor Groups Panel"), DumbAware {
  override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.BGT

  override fun isSelected(event: AnActionEvent): Boolean = state().isShowPanel

  override fun setSelected(event: AnActionEvent, state: Boolean) {
    val editorGroupsSettingsState = state()
    editorGroupsSettingsState.isShowPanel = !editorGroupsSettingsState.isShowPanel

    PanelRefresher.getInstance(event.project!!).refresh()
  }

  companion object {
    const val ID = "krasa.editorGroups.ViewPanelToggleAction"
  }
}
